stats.go

```diff
@@ -46,13 +46,13 @@ func (s *stats) Error() {
 	s.mut.Unlock()
 }
 
-func (s *stats) Reset() stats {
+func (s *stats) Reset() *stats {
 	s.mut.Lock()
 	ns := *s
 	s.announces, s.queries, s.answers, s.errors = 0, 0, 0, 0
 	s.reset = time.Now()
 	s.mut.Unlock()
-	return ns
```

Comment:
This on the other hand is a bona fide bug, thanks for pointing it out. It doesn't hit us due to how the returned stats object is actually used, but it should be fixed nonetheless. I have another approach in mind though.